src/game.cpp

```diff
@@ -5629,11 +5694,23 @@ void Game::playerAcceptMarketOffer(uint32_t playerId, uint32_t timestamp, uint16
 				subType = -1;
 			}
 
-			for (uint16_t i = 0; i < amount; ++i) {
-				Item* item = Item::CreateItem(it.id, subType);
-				if (internalAddItem(player->getInbox(), item, INDEX_WHEREEVER, FLAG_NOLIMIT) != RETURNVALUE_NOERROR) {
-					delete item;
-					break;
+			if (!player->getInbox()) {
+				ItemBlockList inboxDelivery;
+				for (uint16_t i = 0; i < offer.amount; ++i) {
+					Item* item = Item::CreateItem(it.id, subType);
+					if (item) {
+						inboxDelivery.emplace_back(0, item);
+					}
+				}
+				IOInbox::getInstance().pushDeliveryItems(player->getGUID(), inboxDelivery);
```

Comment:
It can be simplified by adding a Game function to add item for player inbox. See my example.